client/src/node/extension.ts

```diff
@@ -49,20 +70,127 @@ export function activate(context: ExtensionContext): void {
     clientOptions
   );
 
+  activeProfileStatusBarIcon.command = "SAS.session.switchProfile";
+
   // Start the client. This will also launch the server
   client.start();
 
   context.subscriptions.push(
     commands.registerCommand("SAS.session.run", run),
-    commands.registerCommand("SAS.session.close", closeSession),
+    commands.registerCommand("SAS.session.close", () => closeSession(false)),
+    commands.registerCommand("SAS.session.switchProfile", switchProfile),
+    commands.registerCommand("SAS.session.addProfile", addProfile),
+    commands.registerCommand("SAS.session.deleteProfile", deleteProfile),
+    commands.registerCommand("SAS.session.updateProfile", updateProfile),
     languages.registerDocumentSemanticTokensProvider(
       { language: "sas-log" },
       LogTokensProvider,
       legend
-    )
+    ),
+    activeProfileStatusBarIcon
+  );
+
+  // First iteration
+  updateStatusBarProfile(activeProfileStatusBarIcon);
+  // Set watcher to update if profile changes
+  activeProfileTracker.activeChanged(async () => {
+    updateStatusBarProfile(activeProfileStatusBarIcon);
+  });
+}
+
+async function updateStatusBarProfile(profileStatusBarIcon: StatusBarItem) {
+  const currentProfile = await profileConfig.getActiveProfile();
+  if (!currentProfile) {
+    resetStatusBarItem(profileStatusBarIcon);
+    return;
+  }
+  updateStatusBarItem(
+    profileStatusBarIcon,
+    `${currentProfile.name}`,
+    `SAS Profile: ${currentProfile.name}\n${currentProfile.profile["sas-endpoint"]}`
   );
 }
 
+function updateStatusBarItem(
+  statusBarItem: StatusBarItem,
+  text: string,
+  tooltip: string
+): void {
+  statusBarItem.text = text;
+  statusBarItem.tooltip = tooltip;
+  statusBarItem.show();
+}
+
+function resetStatusBarItem(statusBarItem: StatusBarItem): void {
+  statusBarItem.text = "No Active Profiles Found";
+  statusBarItem.show();
+}
+
+async function addProfile() {
+  const profileName = await createInputTextBox(ProfilePromptType.NewProfile);
+  if (!profileName) {
+    addProfile();
+    return;
+  }
+  await profileConfig.prompt(profileName).then(() => {
+    activeProfileTracker.setActive(profileName);
+    closeSession(true);
+  });
+}
+
+async function updateProfile() {
+  const profileList = await profileConfig.listProfile();
+  if (profileList.length === 0) {
+    addProfile();
+    return;
+  }
+  const selected = await window.showQuickPick(profileList, {
+    placeHolder: "Update SAS profile",
+  });
+  if (selected) {
+    await profileConfig.prompt(selected, true).then(() => {
+      activeProfileTracker.setActive(selected);
+      closeSession(true);
+    });
+  }
+}
+
+async function switchProfile() {
```

Comment:
Suggest make commands code into src/commands directory, to make extension entry file clean.